### DISCUSSION TITLE
API/Token ability to enable/disable device?

### DISCUSSION THREAD

**Discussion by j4c3** [rustdesk/rustdesk-server-pro#371] [General]
I am trying to write a script to set the Device ID for automated deployment, and need to be able to overwrite an exiting ID.  It didn't seem like that was possible with --set-id, so I was using the API to do it, but, to delete an existing device, I get the response {"error":"Disable it first."}. 

When I try to use {"rows":["GUID"],"disable":true}, I get a response {"error":"Forbidden"}.  Is this expected, or have I missed something?  Is the ability to disable a device not part of the token permission Read/Write for devices?

Also, in checking this, it seems like the web interface is returning an empty results on the Settings -> Tokens page. 

Thanks!

**Comment by rustdesk** [comment]
Please follow https://github.com/rustdesk/rustdesk/blob/master/res/devices.py

**Reply by j4c3** [reply]
Thanks!  I had seen mentioned elsewhere and was trying to pull endpoints out of the network tab on the web interface.
